File: app/src/pfp/java/social/entourage/android/map/entourage/EntourageCloseFragment.java
package social.entourage.android.map.entourage;

import android.content.Context;
import android.content.DialogInterface;
import android.support.v4.app.FragmentManager;
import androidx.appcompat.app.AlertDialog;

import social.entourage.android.R;
import social.entourage.android.api.model.map.FeedItem;
import social.entourage.android.api.tape.Events;
import social.entourage.android.tools.BusProvider;

/**
 * Created by Mihai Ionescu on 03/08/2018.
 */
public class EntourageCloseFragment {

    // ----------------------------------
    // Constants
    // ----------------------------------

    public static final String TAG = EntourageCloseFragment.class.getSimpleName();

    // ----------------------------------
    // Attributes
    // ----------------------------------

    private FeedItem feedItem;

    // ----------------------------------
    // Lifecycle
    // ----------------------------------

    public EntourageCloseFragment() {
        // Required empty public constructor
    }

    public static EntourageCloseFragment newInstance(FeedItem feedItem) {
        EntourageCloseFragment fragment = new EntourageCloseFragment();
        fragment.feedItem = feedItem;
        return fragment;
    }

    public void show(FragmentManager fragmentManager, String tag, Context context) {
        AlertDialog.Builder builder = new AlertDialog.Builder(context);

        builder.setMessage(R.string.entourage_close_alert_description)
                .setPositiveButton(R.string.yes, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(final DialogInterface dialog, final int which) {
                        BusProvider.getInstance().post(new Events.OnFeedItemCloseRequestEvent(feedItem, false, true));
                    }
                })
                .setNegativeButton(R.string.no, null);

        builder.create().show();
    }

}

File: app/src/pfp/java/social/entourage/android/entourage/EntourageCloseFragment.kt
package social.entourage.android.entourage

import android.content.Context
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.FragmentManager
import social.entourage.android.R
import social.entourage.android.api.model.map.FeedItem
import social.entourage.android.api.tape.Events.OnFeedItemCloseRequestEvent
import social.entourage.android.tools.BusProvider

/**
 * Created by Mihai Ionescu on 03/08/2018.
 */
class EntourageCloseFragment  {
    // ----------------------------------
    // Lifecycle
    // ----------------------------------
   private var feedItem: FeedItem? = null
    fun show(fragmentManager: FragmentManager?, tag: String?, context: Context?) {
        val builder = AlertDialog.Builder(context!!)
        builder.setMessage(R.string.entourage_close_alert_description)
                .setPositiveButton(R.string.yes) { dialog, which -> BusProvider.instance.post(OnFeedItemCloseRequestEvent(feedItem, false, true)) }
                .setNegativeButton(R.string.no, null)
        builder.create().show()
    }

    companion object {
        // ----------------------------------
        // Constants
        // ----------------------------------
        val TAG = EntourageCloseFragment::class.java.simpleName

        fun newInstance(feedItem: FeedItem?): EntourageCloseFragment {
            val fragment = EntourageCloseFragment()
            fragment.feedItem = feedItem
            return fragment
        }
    }
}